info.go

```diff
@@ -508,6 +518,34 @@ func (pi *ProgramInfo) VerifiedInstructions() (uint32, bool) {
 	return pi.verifiedInstructions, pi.verifiedInstructions > 0
 }
 
+// KsymAddrs returns the ksym addresses of bpf programs, including subprograms,
+// of current bpf program. The addresses are same as the ones in /proc/kallsyms.
+//
+// Available from 4.18.
+//
+// The bool return value indicates whether this optional field is available.
+func (pi *ProgramInfo) KsymAddrs() ([]uint64, bool) {
+	return pi.ksymInfos, pi.numKsymInfos > 0
+}
+
+type ProgramFunctionInfo = sys.FuncInfo
+
+// FuncInfos returns the function information of bpf programs, including
+// subprograms, of current bpf program.
+//
+// Available from 5.0.
+//
+// The bool return value indicates whether this optional field is available.
+func (pi *ProgramInfo) FuncInfos() ([]ProgramFunctionInfo, bool) {
```

Comment:
I'm curious about the specific use case for this function, since this information is already attached as asm.Instruction metadata to the output of `ProgramInfo.Instructions()`.

This function should at least return something like `[]struct{off uint64, fi btf.FuncInfo}`, otherwise the caller needs to do the work of getting the btf handle and pulling out the right IDs, as demonstrated by the test you added.

Reply:
`[]struct{off uint64, fi btf.FuncInfo}` seems better.

So, `ProgramFunctionInfo` will be:

```Go
type ProgramFunctionInfo struct {
	Off		uint64
	FuncInfo	*btf.Func
}
```

Reply:
Sorry, yes, `btf.FuncInfo` doesn't exist, I meant `btf.Func`. Let's do:
```go
type ProgramFunctionInfo struct {
	InstructionOffset	uint64
	FuncInfo		*btf.Func
}
```

Reply:
> the specific use case for this function

When to run `pwru --output-stack`, it's really helpful to show the exact bpf prog info instead of `bpf_prog_f9e178771bf9c08b_subprog3   [bpf]` for its ksym address, especially for the case there are multiple `bpf_prog_f9e178771bf9c08b_subprog3   [bpf]` for different ksym addresses.

As for my plan, it's to show `subprog3:entry2[type:SCHED_CLS][ID:290][bpf]` for subprog's ksym addresses, `entry2[type:SCHED_CLS][ID:290][bpf]` for non-subprog's ksym addresses. ID is required here because a tc-bpf code can run for multiple pods in k8s env, whose name, tag, type are same but ID.